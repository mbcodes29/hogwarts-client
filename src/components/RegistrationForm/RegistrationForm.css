.RegistrationForm, .user_name, .password, .register_button {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 150px;
    align-self: center;
}