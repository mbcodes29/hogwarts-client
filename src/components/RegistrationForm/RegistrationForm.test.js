import React from 'react';
import ReactDOM from 'react-dom';
import RegistrationForm from './RegistrationForm'
import {BrowserRouter} from 'react-router-dom'
import renderer from "react-test-renderer"

it('renders without crashing',() => {
    const div = document.createElement('div')
    ReactDOM.render(
    <BrowserRouter>
        <RegistrationForm />
    </BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
})

it('renders the UI as expected', () => {
    const tree = renderer
      .create(<BrowserRouter><RegistrationForm /></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });