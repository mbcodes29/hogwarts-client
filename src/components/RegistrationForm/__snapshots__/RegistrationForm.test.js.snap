// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="user_name"
  >
    <label
      htmlFor="RegistrationForm__user_name"
    >
      User Name 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="RegistrationForm__user_name"
      name="user_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="RegistrationForm__password"
    >
      Password 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="RegistrationForm__password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <button
    className="Button register_button"
    type="submit"
  >
    Register
  </button>
</form>
`;
